pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
> As long as everything after -c is treated as positional, that's fine with
> me.  I assume we want $0 to be a fake script in the working directory, so
> that basedir and location hacks work

Passing "/bin/sh" as the first argument makes the script see the same `$0` (`== /bin/sh`) as it would if there were no args.

The hacks you mention are not relevant here, since `$PWD` is the image's `WORKDIR`.